core/src/main/java/org/apache/iceberg/CatalogUtil.java

```diff
@@ -117,4 +122,45 @@ private static void deleteFiles(FileIO io, Set<ManifestFile> allManifests) {
           }
         });
   }
+
+  /**
+   * Load a custom catalog implementation.
+   * The catalog must have a no-arg constructor.
+   * If the catalog implements {@link org.apache.hadoop.conf.Configurable},
+   * {@code Configurable.setConf(org.apache.hadoop.conf.Configuration conf)} is called to set Hadoop configuration.
+   * {@code Catalog.initialize(String name, Map<String, String> options)} is called to complete the initialization.
+   * @param catalogName catalog name
+   * @param impl catalog implementation full class name
+   * @param engineOptions configuration options from a compute engine like Spark or Flink to initialize the catalog
+   * @param hadoopConf hadoop configuration if needed
+   * @return initialized catalog object
+   * @throws IllegalArgumentException if no-arg constructor not found or error during initialization
+   */
+  public static Catalog loadCustomCatalog(
+      String catalogName,
+      String impl,
+      Map<String, String> engineOptions,
```

Comment:
The name `engineOptions` seems too specific because it assumes that the caller is an engine. But it could be a user of the API that isn't an engine. I think `config` or `properties` would be a better name.